cockatrice/src/tab_deck_editor.cpp

```diff
@@ -863,6 +873,43 @@ void TabDeckEditor::actOpenCustomsetsFolder() {
 
 }
 
+void TabDeckEditor::actAddCustomSet()
+{
+
+#if QT_VERSION < 0x050000
+    QString dataDir = QDesktopServices::storageLocation(QDesktopServices::DataLocation);
+#else
+    QString dataDir = QStandardPaths::standardLocations(QStandardPaths::DataLocation).first();
+#endif
+
+    if (!confirmClose())
+        return;
+
+    QFileDialog dialog(this, tr("Load sets/cards"));
+    dialog.setDirectory(dataDir);
+    dialog.setNameFilters(TabDeckEditor::fileNameFilters);
+    if (!dialog.exec())
+        return;
+
+    QString fileName = dialog.selectedFiles().at(0);
+    QDir dir(dataDir.append("/customsets"));
+    QStringList files = dir.entryList();
+    int maxIndex = 0;
+    for (int i = 0; i < files.size(); ++i) {
+        int fileIndex = files.at(i).split(".").at(0).toInt();
```

Comment:
I don't understand what the intent is here. Could you move it to an appropriately named function? Also, I'd strongly prefer more functional style `map` calls and stl iterator interactions instead of raw `for` loops. It's less likely to introduce errors and the declarative style conveys intent a lot more clearly, at least for me.